sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDBStateStoreProvider.scala

```diff
@@ -261,7 +313,21 @@ private[sql] class RocksDBStateStoreProvider
     /** Remove column family if exists */
     override def removeColFamilyIfExists(colFamilyName: String): Boolean = {
       verify(useColumnFamilies, "Column families are not supported in this store")
-      val result = rocksDB.removeColFamilyIfExists(colFamilyName)
+      val result = {
+        val colFamilyExists = ColumnFamilyUtils.checkColFamilyExists(colFamilyName)
+        if (colFamilyExists) {
+          ColumnFamilyUtils.verifyColFamilyOperations("prefixScan", colFamilyName)
+          val idPrefix = ColumnFamilyUtils.getVcfIdBytes(
+            colFamilyNameToIdMap.get(colFamilyName)
+          )
+          rocksDB.prefixScan(idPrefix).foreach { kv =>
+            ColumnFamilyUtils.verifyColFamilyOperations("remove", colFamilyName)
```

Comment:
Is this needed per each state row in the CF? It doesn't seem to be coupled with actual state row at all.

Reply:
yea +1 - i think we don't need this here ?